src/components/discovery/LocusSearch.js

```diff
@@ -1,63 +1,55 @@
-import React, {useEffect, useState} from "react";
-import { AutoComplete } from "antd";
-import { useDispatch, useSelector } from "react-redux";
+import React, { useCallback, useEffect, useState } from "react";
+import { AutoComplete, Tag } from "antd";
 import PropTypes from "prop-types";
-import { performGohanGeneSearchIfPossible } from "@/modules/discovery/actions";
-
-// TODOs:
-// style options
+import { useGeneNameSearch, useReferenceGenomes } from "@/modules/reference/hooks";
 
 const parsePosition = (value) => {
     const parse = /(?:CHR|chr)([0-9]{1,2}|X|x|Y|y|M|m):(\d+)-(\d+)/;
     const result = parse.exec(value);
 
     if (!result) {
-        return {chrom: null, start: null, end: null};
+        return { chrom: null, start: null, end: null };
     }
 
     const chrom = result[1].toUpperCase(); //for eg 'x', has no effect on numbers
     const start = Number(result[2]);
     const end = Number(result[3]);
-    return {chrom, start, end};
+    return { chrom, start, end };
 };
 
 
-const LocusSearch = ({assemblyId, addVariantSearchValues, handleLocusChange, setLocusValidity}) => {
+const LocusSearch = ({ assemblyId, addVariantSearchValues, handleLocusChange, setLocusValidity }) => {
+    const referenceGenomes = useReferenceGenomes();
     const [autoCompleteOptions, setAutoCompleteOptions] = useState([]);
-    const geneSearchResults = useSelector((state) => state.discovery.geneNameSearchResponse);
     const [inputValue, setInputValue] = useState("");
-    const dispatch = useDispatch();
 
-    const showAutoCompleteOptions = assemblyId === "GRCh37" || assemblyId === "GRCh38";
+    const showAutoCompleteOptions = !!referenceGenomes.itemsByID[assemblyId]?.gff3_gz
+        && inputValue.length && (inputValue.includes(" ") || !inputValue.includes(":"));
 
-    const handlePositionNotation = (value) => {
-        const {chrom, start, end} = parsePosition(value);
+    const handlePositionNotation = useCallback((value) => {
+        const { chrom, start, end } = parsePosition(value);
         setLocusValidity(chrom && start && end);
-        addVariantSearchValues({chrom, start, end});
-    };
-
-    const handleChange = (value) => {
+        addVariantSearchValues({ chrom, start, end });
+    }, [setLocusValidity, addVariantSearchValues]);
 
-        setInputValue(value);
-
-        if (!value.includes(" ") && value.includes(":")) {
-            handlePositionNotation(value);
+    useEffect(() => {
+        if (!inputValue.includes(" ") && inputValue.includes(":")) {
```

Comment:
minor: could factor out `inputValue.includes(" ")`, `inputValue.includes(":")` into constants for reuse here and to compute `showAutoCompleteOptions`